tests/utilities/test_auto_restart.py

```diff
@@ -945,23 +945,28 @@ def test_dataset_rng_states_restart_with_lightning(tmpdir, dataset_classes, mult
         multiple_trainloader_mode=multiple_trainloader_mode,
     )
 
-    all_batches = _run_training(trainer_kwargs, dataset_classes)
+    all_batches, weights0 = _run_training(trainer_kwargs, dataset_classes)
     all_batches = torch.stack(all_batches)
     assert len(all_batches) == 9
 
     # Simulate 1st failure
-    complete_batches = _run_training(trainer_kwargs, dataset_classes, fail_on_step=4)
+    complete_batches, _ = _run_training(trainer_kwargs, dataset_classes, fail_on_step=4)
     assert len(complete_batches) == 4
 
     checkpoint_path = os.path.join(tmpdir, ".pl_auto_save.ckpt")
     assert os.path.exists(checkpoint_path)
 
     # Resume after failure
     trainer_kwargs.update(resume_from_checkpoint=checkpoint_path)
-    resumed_batches = _run_training(trainer_kwargs, dataset_classes, fail_on_step=-1)
+    resumed_batches, weights1 = _run_training(trainer_kwargs, dataset_classes, fail_on_step=-1)
     assert len(resumed_batches) == 5
 
     # the resumed batches should match the batches of the successful training
     all_batches_resumed = torch.stack(complete_batches + resumed_batches)
     assert len(all_batches_resumed) == 9
     assert torch.equal(all_batches, all_batches_resumed)
+
+    # the final weights of a resumed training should equal the weights of an uninterrupted training
+    for w0, w1 in zip(weights0, weights1):
+        assert w0 is not w1
+        assert torch.equal(w0, w1)
```

Comment:
Should this maybe use `torch.allclose` instead? Just in case of some floating point weirdness

Reply:
I'm hesitating, because the whole point of fault-tolerant training is that we can restore exactly the state. While there are some limitations to our implementation, this particular test here is simple and all information to restore deterministically should be given.